### ISSUE TITLE
Upload to different locations

### ISSUE THREAD

**Issue by makz27** [lupoDharkael/flameshot#27]
Hi !

Flameshot looks quite promising, i'd love to have the possibility to upload right into a FTP or SFTP and directly get the url copied in my clipboard.

Like puush or the dead project Screencloud.

Do you think you could implement this feature ?

**Comment by lupoDharkael** [comment]
I like the idea!
Thank you for the feedback!

**Comment by Mayurifag** [comment]
+1 here

**Comment by yinflying** [comment]
Good idea, Actually , it is much better to make a button which can run customer shell command.. (then you can upload to anywhere or copy to clipboard, or add a watermarking...) 
：D

**Comment by anarcat** [comment]
this feature request seems to be replicating like crazy! :) there's #50, #178 and #45 already, and, to a lesser extent, #151... It might be worth adding something to the readme regarding that and closing duplicate issues...

in #178, in particular, there is a workaround that uses custom commands to send the captured file to arbitrary locations...

**Comment by de-don** [comment]
I create Go scpirt for uploading screenshot to sftp server.
See it https://github.com/de-don/sftp_uploader

Usage very simple
```
$ flameshot gui -r  | ./sftp_uploader -r

**Comment by viktordw** [comment]
It's been a while, but it seems that this feature is still not implemented. What are your plans on this @mmahmoudian @hosiet?

**Comment by mmahmoudian** [comment]
@viktordw it has been discussed many times in various form, and we always got to one conclusion: upload is not a screenshot functionality. We already have imgur which is from time to time pain to maintain due to changes from imgur API or quota issues. For now we will keep the imgur as only natively supported platform, and every other means of file transfer over network should be done via plugins. You can find a list of all stuff marked for plugins under [plugin label](https://github.com/flameshot-org/flameshot/issues?q=is%3Aopen+is%3Aissue+label%3APlugin). At the time of writing this the plugin backend is not implemented, and you can follow its progress on #2529 .

Also as always, in the spirit of opensource project, any contribution (as long as it is aligned with the collective vision of the devs) is more than welcome. Also if you cannot contribute via code, design, or documentation, you can always put bounty on the issues you want to be addressed to create more incentive for the community to contribute. More info here:

https://flameshot.org/donate/
